src/restic/node.go

```diff
@@ -378,7 +409,48 @@ func (node Node) sameContent(other Node) bool {
 			return false
 		}
 	}
+	return true
+}
 
+func (node Node) sameAcl(other Node) bool {
+	if node.Xattrs == nil {
+		return other.Xattrs == nil
+	}
+	if other.Xattrs == nil {
+		return false
+	}
+	if len(node.Xattrs) != len(other.Xattrs) {
+		return false
+	}
+	for _, thisAttr := range node.Xattrs {
```

Comment:
Split this into its own function that compares two `[]Attribute` slices by building a `map[Attribute]struct{}` and comparing those, see https://github.com/restic/restic/blob/master/src/restic/idset.go#L47 for a sample on the `IDSet` type (which is just a `[]ID`).